src/components/task-plan/homework.cjsx

```diff
@@ -0,0 +1,404 @@
+    content = JSON.parse(@props.exercise.content)
+    question = content.questions[0]
+    renderedAnswers = _.map(question.answers, @renderAnswers)
+    renderedTags = _.map(content.tags, @renderTags)
+
+    header = @renderHeader()
+    panelStyle = @getPanelStyle()
+
+    <BS.Panel bsStyle={panelStyle} header={header}>
+      <ArbitraryHtmlAndMath className="-stimulus" block={true} html={content.stimulus_html} />
+      <ArbitraryHtmlAndMath className="-stem" block={true} html={question.stem_html} />
+      <ul>{renderedAnswers}</ul>
+      <p>{renderedTags}</p>
+    </BS.Panel>
+
+ReviewExerciseCard = React.createClass
+  mixins: [ExerciseCardMixin]
+
+  moveExerciseUp: ->
+    TaskPlanActions.moveExercise(@props.planId, @props.exercise, -1)
+
+  moveExerciseDown: ->
+    TaskPlanActions.moveExercise(@props.planId, @props.exercise, 1)
+
+  removeExercise: ->
+    TaskPlanActions.removeExercise(@props.planId, @props.exercise)
+
+  renderHeader: ->
+    <span>
+      <span class="-exercise-number">{@props.index+1}</span>
+      <BS.Button bsStyle="info" onClick={@moveExerciseUp} className="-move-exercise-up">^</BS.Button>
+      <BS.Button bsStyle="info" onClick={@moveExerciseDown} className="-move-exercise-up">v</BS.Button>
+      <BS.Button bsStyle="danger" onClick={@removeExercise} className="-remove-exercise-up">X</BS.Button>
+    </span>
+
+  getPanelStyle: ->
+    "default"
+
+  render: ->
+    @renderExercise()
+
+
+AddExerciseCard = React.createClass
+  mixins: [ExerciseCardMixin]
+
+  toggleExercise: ->
+    if TaskPlanStore.hasExercise(@props.planId, @props.exercise.id)
+      TaskPlanActions.removeExercise(@props.planId, @props.exercise)
+    else
+      TaskPlanActions.addExercise(@props.planId, @props.exercise)
+
+  renderHeader: ->
+    active = TaskPlanStore.hasExercise(@props.planId, @props.exercise.id)
+    toggleText = if not active then <span>+</span> else <span>-</span>
+    <span><BS.Button bsStyle="primary" onClick={@toggleExercise} className="-add-exercise">{toggleText}</BS.Button></span>
+
+  getPanelStyle: ->
+    if TaskPlanStore.hasExercise(@props.planId, @props.exercise.id)
+      return "info"
+    else
+      return "default"
+
+  render: ->
+    @renderExercise()
+
+ExercisesRenderMixin =
+  componentWillMount:   -> ExerciseStore.addChangeListener(@update)
+  componentWillUnmount: -> ExerciseStore.removeChangeListener(@update)
+
+  update: ->
+    @setState({})
+
+
+  renderLoading: ->
+    {courseId, pageIds, hide} = @props
+
+    unless @props.shouldShow
+      return <span className="-no-show"/>
+    unless ExerciseStore.isLoaded(pageIds)
+      ExerciseActions.load(courseId, pageIds)
+      return <span className="-loading">Loading...</span>
+    
+    false
+
+
+ReviewExercises = React.createClass
+  mixins: [ExercisesRenderMixin]
+
+  renderExercise: (exercise, i) ->
+    <ReviewExerciseCard index={i} planId={@props.planId} exercise={exercise}/>
+
+  render: ->
+    load = @renderLoading()
```

Comment:
maybe won't work, but an idea -- worth it to alias ExerciseActions.load(courseId, pageIds) to ExerciseActionByCourse(pageIds) with currying/using http://underscorejs.org/#partial ?